zlint.go

```diff
@@ -0,0 +1,50 @@
+/* zlint.go
+ * Used to check parsed info from certificate for compliance
+ */
+
+package zlint
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"time"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/lints"
+)
+
+type LintDescription struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Provenance  string `json:"provenance"`
+}
+
+// EncodeLintDescriptionsToJSON outputs a description of the lint as JSON.
+func EncodeLintDescriptionsToJSON(w io.Writer) {
+	for _, l := range lints.Lints {
+		p := LintDescription{}
+		p.Name = l.Name
+		p.Description = l.Description
+		p.Provenance = l.Provenance
+
+		buffer := new(bytes.Buffer)
+		enc := json.NewEncoder(buffer)
+		enc.SetEscapeHTML(false)
+		enc.Encode(p)
+		w.Write(buffer.Bytes())
+	}
+}
+
+// LintCertificate runs all registered lints on c, producing a ZLintResult.
+func LintCertificate(c *x509.Certificate) *lints.ZLintResult {
+	if c == nil {
```

Comment:
This seems like an odd path offhand.

Reply:
I don't like this function either, I just copied it over from the old version and gave it a better name. It should really return an error meaning "zlint could not run for some reason that is not lint related", but I think that's outside of the scope of this PR.